harness/features/allFeatures.cloud.test.ts

```diff
@@ -0,0 +1,88 @@
+    forEachSDK, 
+    describeIf, 
+    createUser,
+    TestClient,
+    wait,
+} from '../helpers'
+import { Capabilities, SDKCapabilities } from '../types'
+import { v4 as uuidv4 } from 'uuid'
+import { getServerScope } from '../nock'
+import { expectedFeaturesVariationOn } from '../mockData'
+
+jest.setTimeout(10000)
+
+const scope = getServerScope()
+
+describe('allFeatures Tests - Cloud', () => {
+    forEachSDK((name) => {
+        const testClient = new TestClient(name)
+        
+        let url: string
+        const capabilities: string[] = SDKCapabilities[name]
+
+        let variationOnUser: string
+        let invalidUser: string
+    
+        beforeAll(async () => {
+            url = getConnectionStringForProxy(name)  
+            await testClient.createClient({ 
+                enableCloudBucketing: true,
+            })
+    
+            variationOnUser = (
+                await createUser(url, { user_id: 'user1', customData: { 'should-bucket': true } })
+            ).headers.get('location')
+    
+            invalidUser = (
+                await createUser(url, { name: 'invalid' })
+            ).headers.get('location')
+        })
+
+        describeIf(capabilities.includes(Capabilities.cloud))(name, () => {
+            it('should return all features without edgeDB', async () => {
+                scope
+                    .post(`/${testClient.clientLocation}/v1/features`, (body) => body.user_id === 'user1')
+                    .matchHeader('Content-Type', 'application/json')
+                    .matchHeader('authorization', testClient.sdkKey)
+                    .query((queryObj) => {
+                        return !queryObj.enableEdgeDB
+                    })
+                    .reply(200, expectedFeaturesVariationOn)
+                const featuresResponse = await testClient.callAllFeatures(variationOnUser)
+                
+                const features = (await featuresResponse.json()).data
+                expect(features).toMatchObject(expectedFeaturesVariationOn)
+            })
+
+            it('should return all features with edgeDB', async () => {
+                const edgeDBTestClient = new TestClient(name)
+                await edgeDBTestClient.createClient({ 
+                    enableEdgeDB: true,
+                    enableCloudBucketing: true,
+                })
+
+                scope
+                    .post(`/${edgeDBTestClient.clientLocation}/v1/features`)
+                    .matchHeader('Content-Type', 'application/json')
+                    .matchHeader('authorization', edgeDBTestClient.sdkKey)
+                    .query((queryObj) => {
+                        return !!queryObj.enableEdgeDB
+                    })
+                    .reply(200, expectedFeaturesVariationOn)
+                
+                const featuresResponse = await edgeDBTestClient.callAllFeatures(variationOnUser)
+                const features = (await featuresResponse.json()).data
```

Comment:
might as well test the entityType of the response here as well?

Reply:
good point, added